lib/util.js

```diff
@@ -193,3 +195,52 @@ function encode_uri(uri) {
 }
 
 exports.encode_uri = encode_uri;
+
+function fetch_rss_entries (feed_url, max_items=100, max_days_old=7) {
+    const oldest_date = moment().subtract(max_days_old, 'days');
+
+    return _fetch_rss_page(feed_url, 1, [], max_items, oldest_date);
+}
+
+function _fetch_rss_json (feed_url) {
+    return new Promise((resolve, reject) => {
+        const req = request.get(feed_url);
+        const parser = new feedparser();
+        req.pipe(parser);
+
+        req.on('error', reject);
+        parser.on('error', reject);
+
+        const feed_data = { items: [] };
+        parser.on('meta', meta => feed_data.meta = meta);
+        parser.on('data', article => feed_data.items.push(article));
+        parser.on('end', () => resolve(feed_data));
+    });
+}
+
+// Assume a wordpress feed (https://codex.wordpress.org/Pagination#Adding_the_.22paged.22_parameter_to_a_query)
+function _fetch_rss_page (feed_url, page, items, max_items, oldest_date) {
+    let paginated_url = feed_url;
+    if (page > 1) {
+        paginated_url += `?paged=${page}`;
+    }
+
+    return _fetch_rss_json(paginated_url).then(feed_json => {
+        const recent_enough_items = feed_json.items.filter(article => article.pubdate >= oldest_date);
+        // If we've run into articles which are too old, cease crawling
+        if (recent_enough_items.length !== feed_json.items.length) {
+            return items.concat(recent_enough_items);
+        }
+
+        const new_items = items.concat(feed_json.items);
+        // If we've hit the desired number of items, also cease crawling
+        if (new_items.length >= max_items) {
+            return new_items.slice(max_items);
+        }
```

Comment:
Why not just concat the feed list directly with slice and then check the length which would prevent the need for `new_items` var completely?